src/lib/swapper/swappers/CowSwapper/CowSwapper.ts

```diff
@@ -27,7 +27,10 @@ export const cowSwapper: Swapper = {
     // Some context about this : https://docs.cow.fi/tutorials/how-to-submit-orders-via-the-api/4.-signing-the-order
     // For more info, check hashOrder method implementation
     const orderDigest = hashOrder(domain(signingDomain, COW_SWAP_SETTLEMENT_ADDRESS), orderToSign)
-    const messageToSign = ethers.utils.arrayify(orderDigest)
+    // orderDigest should be an hex string here. All we need to do is pass it to signMessage/wallet.ethSignMessage and sign it
```

Comment:
An hex string is, as in hex as a string, not a string as hex - see https://docs.cow.fi/smart-contracts/settlement-contract/signature-schemes
This is effectively an hashed EIP-712 struct

Reply:
Perhaps https://docs.ethers.org/v5/api/utils/bytes/#utils-isHexString to be safe here